packages/block-library/src/query/index.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+/**
+ * Server-side rendering of the `core/query` block.
+ *
+ * @package WordPress
+ */
+class Blocks_Query {
+	/**
+	 * @var array
+	 */
+	public static $displayedPostIds = array();
+}
+
+/**
+ * Registers the `core/query` block on server.
+*/
+function register_block_core_query() {
+	register_block_type(
+		'core/query',
+		array(
```

Comment:
Seems like the attributes defined on the server and client are different. Should we just define all the attributes here and remove the client ones?

Reply:
I tested it, and [block-library/src/index.js](https://github.com/WordPress/gutenberg/blob/b3cc946c41bdcf00262e0bafdfe8f7bf779eb581/packages/block-library/src/index.js#L90-L92) overwrites the server-side attribute definition if a block exports `metadata`.  For now, I'll just make sure the client and server definitions match.

Reply:
@gziolo may know more how to do it right. We have some dynamic blocs already that only define the server attributes I think.

Reply:
Yes, sure. There is Shortcode block:

https://github.com/WordPress/gutenberg/blob/099929ba87d61403b5b0c30571f58c70954b387e/packages/block-library/src/shortcode/index.php#L24-L34

Reply:
Thanks @gziolo, that worked